src/main/java/com/redhat/pantheon/servlet/ModuleRevisionUpload.java

```diff
@@ -104,6 +110,12 @@ protected void doRun(SlingHttpServletRequest request, PostResponse response, Lis
                     .content.getOrCreate()
                     .asciidoc.getOrCreate()
                     .jcrContent.getOrCreate();
+            boolean generateHtml = false;
+            String jcrData = jcrContent.jcrData.get();
+
+            if ((jcrData != null && !jcrData.equals(asciidocContent)) || draftRevision.get().content.get().cachedHtml.get() == null) {
```

Comment:
There's a danger here that any of those `.get()` calls returns a null, in which case the whole chain of calls should be considered as a null. This is an upgrade to the framework I have on a branch. Might be a good time to submit it.

Reply:
The changes I mentioned above have already been integrated. The null-safe way to execute these set of calls and to validate if there is cached html already is as follows:

```java
draftRevision.map(i -> i.content.get()).map(i -> i.cachedHtml.get()).isPresent()
```
(I know it's not the most readable way, but blame java for its verboseness)
This would guarantee that at any point in the call if there is a missing child in the tree, this would return false, without any NPEs thrown.